graylog2-server/src/main/java/org/graylog2/security/hashing/BCryptPasswordAlgorithm.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog2.security.hashing;
+
+import com.google.common.base.Splitter;
+import org.graylog2.plugin.security.PasswordAlgorithm;
+import org.mindrot.jbcrypt.BCrypt;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import java.util.List;
+
+public class BCryptPasswordAlgorithm implements PasswordAlgorithm {
+    private static final String PREFIX = "{bcrypt}";
+    private static final String SALT_PREFIX = "{salt}";
+
+    private final Integer saltSize;
+
+    @Inject
+    public BCryptPasswordAlgorithm(@Named("user_password_bcrypt_salt_size") Integer saltSize) {
+        this.saltSize = saltSize;
+    }
+
+    @Override
+    public boolean supports(String hashedPassword) {
+        return hashedPassword.startsWith(PREFIX);
+    }
+
+    private String hash(String password, String salt) {
+        return PREFIX + BCrypt.hashpw(password, salt) + SALT_PREFIX + salt;
+    }
+
+    @Override
+    public String hash(String password) {
+        return hash(password, BCrypt.gensalt(this.saltSize));
+    }
+
+    @Override
+    public boolean matches(String hashedPasswordAndSalt, String otherPassword) {
+        if (supports(hashedPasswordAndSalt)) {
+            final Splitter splitter = Splitter.on(SALT_PREFIX);
```

Comment:
Instead of using `Splitter` here, I'd use standard string operations here, for example:

```java
        final int saltIndex = hashedPasswordAndSalt.lastIndexOf(SALTPREFIX);
        checkArgument(saltIndex > 0, "Couldn't find index of salt prefix");

        final String salt = hashedPasswordAndSalt.substring(saltIndex + SALTPREFIX.length());

        return hash(otherPassword, salt).equals(hashedPasswordAndSalt);
```
Additionally, the current implementation will fail with an `IndexOutOfBoundsException` if the `SALT_PREFIX` is missing (which should rather be an `IllegalArgumentException`.